File: cholericdev.model/Post.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace cholericdev.model
{
    public class Post
    {
        public int PostId { get; private set; }
        public string Title { get; set; }
        public Category Category { get; set; }
        public string Content { get; set; }

        private DateTime creationDay;
        public DateTime CreationDay
        {
            get => creationDay;
            set => creationDay = value;
        }

        private DateTime lastUpdate;
        public DateTime LastUpdate
        {
            get => lastUpdate;
            set => lastUpdate = value;
        }

        public User User { get; set; }
        public ICollection<Comment> Comments { get; set; }
    }
}

File: cholericdev.web/ClientApp/src/app/shared/post.ts
import { Comment } from './comment';
import { User } from './user';
import { Category } from './category';

export interface Post {
  postId: number;
  title: string;
  category: Category;
  content: string;
  creationDay: Date;
  lastUpdate: Date;
  user: User;
  comments: Comment[];
}

